client/src/client.ts

```diff
@@ -2759,6 +2759,15 @@ export abstract class BaseLanguageClient {
 				let actions = params.actions || [];
 				return messageFunc(params.message, ...actions);
 			});
+			connection.onRequest(ShowTextDocumentRequest.type, async (params) => {
+				try {
+					const { textDocument: { uri }, options } = params;
+					const document = await Workspace.openTextDocument(Uri.parse(uri));
+					return Boolean(await Window.showTextDocument(document, this.protocol2CodeConverter.asTextDocumentShowOptions(options)));
```

Comment:
I've changed the return type to `boolean` to give the server some knowledge about the client successfully opening the document or not.
I don't know what's the best way to accomplish this with the language server protocol (maybe returning the error would be better?), so feedback is welcome.

Reply:
What do you anticipate servers doing if the client fails to open the file?

Reply:
JSON-RPC supports returning either ok response or an error natively. There's no need to make the return type here non-void, the client can communicate the error using the protocol directly

See https://microsoft.github.io/language-server-protocol/specification, section `Response Message`.

Reply:
Just to follow-up on why I said what I said, I was wondering if this could be a notification instead of a response? It's not clear to me what a server would do in response to a client that didn't open the file.

Reply:
Thanks, I'll simply take advantage of the JSON-RPC capability to natively return error messages then.